### PR TITLE
adds support for a custom user-agent

### PR DESCRIPTION
**Pull request by alexandreio**
This pull request solve the problem with a custom user-agent described on the issue #145 

Example of the usage:
  tabula.read_pdf(uri, user_agent='Mozilla/5.0')

### CODE REVIEW

File: tests/test_read_pdf_table.py [hunk 1]
```diff
@@ -30,6 +30,11 @@ def test_read_remote_pdf(self):
         df = tabula.read_pdf(uri)
         self.assertTrue(isinstance(df, pd.DataFrame))
 
+    def test_read_remote_pdf_with_custom_user_agent(self):
+        uri = "https://github.com/tabulapdf/tabula-java/raw/master/src/test/resources/technology/tabula/12s0324.pdf"
+        df = tabula.read_pdf(uri, user_agent='Mozilla/5.0')
+        self.assertTrue(isinstance(df, pd.DataFrame))
```
Comment (chezou) [inline tests/test_read_pdf_table.py]: It seems this test doesn't check user agent. Would it be possible to check it?

Comment (alexandreio) [reply tests/test_read_pdf_table.py]: I'm trying to figure out a way to test it. Can I create a local server using flask for this purpose. Or do you have some better ideia?

Comment (chezou) [reply tests/test_read_pdf_table.py]: IMHO, That's too heavy for this test.

I thought Mocking and validating `urlopen` with the expected `Request` object can be fine for it. https://docs.python.org/3/library/unittest.mock.html#quick-guide 
How about this approach? Or, if it doesn't work for it, it's ok to keep as is.

Comment (alexandreio) [reply tests/test_read_pdf_table.py]: I tried this without success. I create a new function with the functionality to create the request and call it on the tests

Comment (chezou) [reply tests/test_read_pdf_table.py]: I mean I think you can write a test with `patch` like this https://github.com/chezou/tabula-py/blob/master/tests/test_read_pdf_table.py#L182-L191

It's OK to create a new function but `assert_called_with` is enough in this case.

Comment (alexandreio) [reply tests/test_read_pdf_table.py]: I tried here. Didn't work. I'll try to figure out still this week how to test with the `assert_called_with`

**Comment by chezou** [comment]
Thanks for your contribution, @alexandreio ! I modified based on your PR at #151

**Comment by alexandreio** [comment]
> Thanks for your contribution, @alexandreio ! I modified based on your PR at #151

I'll try to contribute more with this project on the future :D

**Pull request merged**